client/verta/verta/_internal_utils/_utils.py

```diff
@@ -68,19 +70,111 @@ def __init__(self, scheme=None, socket=None, auth=None, max_retries=0, ignore_co
             on HTTP codes {502, 503, 504} which commonly occur during back end connection lapses.
         ignore_conn_err : bool, default False
             Whether to ignore connection errors and instead return successes with empty contents.
+        credentials : :class:`~verta._internal_utils.credentials.EmailCredentials` or :class:`~verta._internal_utils.credentials.JWTCredentials`, optional
+            Either dev key or JWT token data to be used for authentication.
+        headers: dict, optional
+            Additional headers to attach to requests.
 
         """
+        self._init_headers()
         self.scheme = scheme
         self.socket = socket
-        self.auth = auth
         # TODO: retry on 404s, but only if we're sure it's not legitimate e.g. from a GET
         self.retry = Retry(total=max_retries,
                            backoff_factor=1,  # each retry waits (2**retry_num) seconds
                            method_whitelist=False,  # retry on all HTTP methods
-                           status_forcelist=(502, 503, 504),  # only retry on these status codes
+                           status_forcelist=(requests.codes.bad_gateway, requests.codes.unavailable, requests.codes.gateway_timeout),  # only retry on these status codes
                            raise_on_redirect=False,  # return Response instead of raising after max retries
                            raise_on_status=False)  # return Response instead of raising after max retries
         self.ignore_conn_err = ignore_conn_err
+        self.credentials = credentials
+        self.headers = headers
+
+
+    @property
+    def credentials(self):
+        return self._credentials
+
+    @credentials.setter
+    def credentials(self, value):
+        self._credentials = value
+        self._recompute_headers()
+
+    @property
+    def headers(self):
+        return self._computed_headers
+
+    # Note: Added for temporary backwards compatibility. Remove when possible.
+    @property
+    def auth(self):
+        return self.headers
+
+    @headers.setter
+    def headers(self, value):
+        if value:
+            self._headers = value
+        else:
+            self._headers = dict()
+        self._recompute_headers()
+
+    def _init_headers(self):
+        self._headers = {}
+        self._computed_headers = {}
+
+    def _recompute_headers(self):
+        headers = self._headers or dict()
+        headers = headers.copy()
+        headers[_GRPC_PREFIX+'scheme'] = self.scheme
+        headers.update(self._prefixed_headers_for_credentials(self.credentials))
+        self._computed_headers = headers
+
+    def _prefixed_headers_for_credentials(self, credentials):
```

Comment:
No strong preference on this one.
```suggestion
    @staticmethod
    def _prefixed_headers_for_credentials(credentials):
```